clientd3d/ownerdrw.c

```diff
@@ -68,21 +68,59 @@ void WindowEndUpdate(HWND hwnd)
  */
 int OwnerListAddItem(HWND hwnd, object_node *obj, int index, Bool combo, Bool quan)
 {
-   char *name, desc[MAXNAME + 15];
+   char suffix[MAXNAME + 15];
+   char nameWithSuffix[MAXNAME + 15];
+   char nameWithQuantity[MAXNAME + 15];
+   char *name;
    int pos;
 
    name = LookupNameRsc(obj->name_res);
-   /* If item is a number object, add its amount after the item's name */
+
+   // Suffix the name based on the object's rarity
+   switch(obj->rarity) 
+   {
+   case ITEM_RARITY_GRADE_NORMAL:
+      strcpy(suffix, ""); // Empty string
+      break;
+   case ITEM_RARITY_GRADE_MAGIC:
+      strcpy(suffix, " (magic)");
```

Comment:
This duplicates all of the names from the previous code.  That's a recipe for problems in the future.

Instead, make a single function that takes a rarity and returns a string.

Also, to make future translation possible, move these text strings to the .rc file.

Reply:
Done
- Created new function `GetRaritySuffix(int rarity)` which returns a string as requested.
- Moved text strings to .rc file and referenced appropriately